### PR TITLE
refactor(log): use LOG_ERROR_F instead of LOG_ERROR (1/2)

### PR DESCRIPTION
**Pull request by WHBANG**
### What problem does this PR solve? 
https://github.com/apache/incubator-pegasus/issues/1305

This patch aim to use LOG_ERROR_F instead of LOG_ERROR. Includes:

Use LOG_ERROR_F instead of LOG_ERROR in replica module.

### CODE REVIEW

File: src/replica/replica_stub.cpp [hunk 1]
```diff
@@ -1633,15 +1631,15 @@ void replica_stub::response_client(gpid id,
...
-                  request == nullptr ? "null" : request->header->from_address.to_string(),
-                  request == nullptr ? "null" : request->header->rpc_name,
-                  request == nullptr ? 0 : request->header->client.timeout_ms,
-                  enum_to_string(status),
-                  error.to_string());
+        LOG_ERROR_F("{}@{}: {} fail: client = {}, code = {}, timeout = {}, status = {}, error = {}",
+                    id,
+                    _primary_address_str,
+                    is_read ? "read" : "write",
+                    request == nullptr ? "null" : request->header->from_address,
```
[the remark this answers is not in the document]
Comment (WHBANG) [reply src/replica/replica_stub.cpp]: done

File: src/replica/replica_restore.cpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ bool replica::remove_useless_file_under_chkpt(const std::string &chkpt_dir,
     // filename --> file_path such as: file --> ***/***/file
     std::map<std::string, std::string> name_to_filepath;
     if (!::dsn::utils::filesystem::get_subfiles(chkpt_dir, sub_files, false)) {
-        LOG_ERROR("%s: get subfile of dir(%s) failed", name(), chkpt_dir.c_str());
+        LOG_ERROR_PREFIX("get subfile of dir(%s) failed", chkpt_dir);
```
[the remark this answers is not in the document]
Comment (WHBANG) [reply src/replica/replica_restore.cpp]: done

File: src/replica/log_file.cpp [hunk 3]
```diff
@@ -203,27 +202,27 @@ error_code log_file::read_next_log_block(/*out*/ ::dsn::blob &bb)
             // if read_count is 0, then we meet the end of file
             err = (bb.length() == 0 ? ERR_HANDLE_EOF : ERR_INCOMPLETE_DATA);
         } else {
-            LOG_ERROR("read data block header failed, size = %d vs %d, err = %s",
-                      bb.length(),
-                      (int)sizeof(log_block_header),
-                      err.to_string());
+            LOG_ERROR_F("read data block header failed, size = {} vs {}, err = {}",
+                        bb.length(),
+                        (int)sizeof(log_block_header),
```
[the remark this answers is not in the document]
Comment (WHBANG) [reply src/replica/log_file.cpp]: done

File: src/replica/log_file.cpp [hunk 4]
```diff
@@ -203,27 +202,27 @@ error_code log_file::read_next_log_block(/*out*/ ::dsn::blob &bb)
...
 
     err = _stream->read_next(hdr.length, bb);
     if (err != ERR_OK || hdr.length != bb.length()) {
-        LOG_ERROR("read data block body failed, size = %d vs %d, err = %s",
-                  bb.length(),
-                  (int)hdr.length,
-                  err.to_string());
+        LOG_ERROR_F("read data block body failed, size = {} vs {}, err = {}",
+                    bb.length(),
+                    (int)hdr.length,
```
[the remark this answers is not in the document]
Comment (WHBANG) [reply src/replica/log_file.cpp]: done

File: src/replica/replica_2pc.cpp [hunk 5]
```diff
@@ -524,10 +518,8 @@ void replica::on_append_log_completed(mutation_ptr &mu, error_code err, size_t s
     if (err == ERR_OK) {
         mu->set_logged();
     } else {
-        LOG_ERROR("%s: append shared log failed for mutation %s, err = %s",
-                  name(),
-                  mu->name(),
-                  err.to_string());
+        LOG_ERROR_PREFIX(
+            "append shared log failed for mutation {}, err = {}", mu->name(), err.to_string());
```
[the remark this answers is not in the document]
Comment (WHBANG) [reply src/replica/replica_2pc.cpp]: done

**Review by empiredan — APPROVED** [review]

File: src/replica/replica_learn.cpp [hunk 6]
```diff
@@ -1227,22 +1213,19 @@ error_code replica::handle_learning_succeeded_on_primary(::dsn::rpc_address node
...
 
     if (it->second.signature != (int64_t)learn_signature) {
-        LOG_ERROR("%s: handle_learning_succeeded_on_primary[%016" PRIx64 "]: learner = %s, "
-                  "signature not matched, current signature on primary is [%016" PRIx64
-                  "], return ERR_INVALID_STATE",
-                  name(),
-                  learn_signature,
-                  node.to_string(),
-                  it->second.signature);
+        LOG_ERROR_F("handle_learning_succeeded_on_primary[{:#018x}]: learner = {}, signature not "
```
Comment (acelyc111) [inline src/replica/replica_learn.cpp:1224]: Should it be `LOG_ERROR_PREFIX`?

Comment (WHBANG) [reply src/replica/replica_learn.cpp]: fixed

**Review by empiredan — APPROVED** [review]

**Review by acelyc111 — APPROVED** [review]

**Pull request merged**